lib/transaction.js

```diff
@@ -117,47 +117,49 @@ class Transaction {
       connectionPromise = this.sequelize.connectionManager.getConnection(acquireOptions);
     }
 
-    return connectionPromise
-      .then(connection => {
-        this.connection = connection;
-        this.connection.uuid = this.id;
-      })
-      .then(() => {
-        return this.begin()
-          .then(() => this.setDeferrable())
-          .catch(setupErr => this.rollback().finally(() => {
-            throw setupErr;
-          }));
-      })
-      .then(result => {
-        if (useCLS && this.sequelize.constructor._cls) {
-          this.sequelize.constructor._cls.set('transaction', this);
-        }
-        return Promise.resolve(null).then(() => result);
-      });
+    let result;
+    const connection = await connectionPromise;
+    this.connection = connection;
+    this.connection.uuid = this.id;
+
+    try {
+      await this.begin();
+      result = await this.setDeferrable();
+    } catch (setupErr) {
+      try {
+        result = await this.rollback();
+      } finally {
+        throw setupErr; // eslint-disable-line no-unsafe-finally
+      }
+    }
+
+    if (useCLS && this.sequelize.constructor._cls) {
+      this.sequelize.constructor._cls.set('transaction', this);
+    }
+
+    return result;
   }
 
-  setDeferrable() {
+  async setDeferrable() {
     if (this.options.deferrable) {
-      return this
+      return await this
```

Comment:
Nit: await seems redundant here.

Reply:
Oh weird, that was definitely not intentional, I had intended for my codemod to never await this 😅

Reply:
Oh I see there's stuff after it.  Yeah I went back and forth on whether to `return await` promises or just return them.  In the end I decided to `return await` in most cases to make it clear the method I was calling is async, and in case someone makes changes where they remove the `return` so they can add more statements after it, but forget to add `await`.

However, my codemod doesn't try to determine whether a method call is sync or async so there may be a few unnecessary `await`s, which are harmless but not ideal.  (I tried to look through the changes and get rid of `await` on any known sync calls).

In any case, removing `await` from return statements (caution: as long as they aren't in a try block) would be easy to do with a codemod, do you want me to make that change everywhere?

Reply:
I think it's best to keep `return await` the exception for try blocks.
I'd defer fixing existing code until the async/await conversion is done :)